src/lib/ecore/efl_promise2.h

```diff
@@ -0,0 +1,32 @@
+#ifndef _EFL_PROMISE2_H_
+#define _EFL_PROMISE2_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <Eina.h>
+
+typedef struct _Efl_Promise2 Efl_Promise2;
+typedef struct _Efl_Future2 Efl_Future2;
+typedef Eina_Value *(*Efl_Future2_Cb)(void *data, const Eina_Value *value);
+
+EAPI Efl_Promise2 *efl_promise2_new(Eina_Free_Cb cancel_cb, const void *data);
+EAPI Eina_Bool efl_promise2_resolve(Efl_Promise2 *p, Eina_Value *value);
+EAPI Eina_Bool efl_promise2_reject(Efl_Promise2 *p, Eina_Value *value);
```

Comment:
btw, since we're adopting `Eina_Error` for rejection, then we can use the signature:

`efl_promise2_reject(Efl_Promise2 *p, Eina_Error error);`

Do not return errors here and for `efl_promise2_resolve()`, on failure always free the given value.